flow/core/kernel/vehicle/traci.py

```diff
@@ -970,7 +970,7 @@ def apply_lane_change(self, veh_ids, direction):
             # perform the requested lane action action in TraCI
             if target_lane != this_lane:
                 self.kernel_api.vehicle.changeLane(
-                    veh_id, int(target_lane), 100000)
+                    veh_id, int(target_lane), self.sim_step)
```

Comment:
This is the result of a strange lane changing bug that took me quite some time to debug. With both custom lane changing models and RL lane changing models, the controller issues a lane change command (change left, change right, stay in lane) at every timestep. However, we only send traci commands if we're actually executing a lane change (see line 958).

The behavior we want to enforce is that the lane changing model's current action is correct. Before this change, we would get the following behavior:

1. the model is requesting a lane change
2. traci would block it because it was unsafe
3. model stops requesting a lane change
4. because our past lane change is still active and we do not update the lane change, the vehicle changes lanes
 4.5 i believe the reason we do not explicitly update all vehicle lane changes is to speed things up? or maybe because we don't want to overwrite the sumo lane changing model
@AboudyKreidieh @eugenevinitsky @kjang96  let me know what y'all think about this

Reply:
@AboudyKreidieh, this seems fine to me?

Reply:
seems fine to me too